app/controllers/codelists-management/codelist.js

```diff
@@ -0,0 +1,40 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class CodelistController extends Controller {
+  @service router;
+  @tracked isOpen = false;
+
+  get hasActiveChildRoute() {
+    return (
+      this.router.currentRouteName.startsWith(
+        'codelists-management.codelist'
+      ) &&
+      this.router.currentRouteName !== 'codelists-management.codelist.index'
+    );
+  }
+
+  @action
+  toggleIsOpen() {
+    this.isOpen = !this.isOpen;
+  }
+
+  @action
+  async removeCodelist(codelist, event) {
+    event.preventDefault();
+
+    const length = codelist.codeListOptions.length;
+    for (let i = 0; i < length; i++) {
+      const option = codelist.codeListOptions.objectAt(0);
+      await option.destroyRecord();
+    }
```

Comment:
Something like this makes it a bit shorter (and in my opinion more readable)
```suggestion
    await Promise.all(codelist.codeListOptions.map((option) => option.destroyRecord()));
```

Reply:
I do wonder why all of these need to be destroyed (and partly) recreated on save though. Wouldn't it be better to mark them for deletion and only do the actual deletion when needed?

Reply:
How would you `mark them for deletion` ? I actually did it that way because I didn't see how I could do what you suggest haha

Reply:
If you use the array with records instead of the array with strings as the options for the table I think you can use [`deleteRecord`](https://api.emberjs.com/ember-data/3.28/classes/Model/methods/deleteRecord?anchor=deleteRecord) when clicking the delete button? I'm not sure if it will automatically be filtered from the array but if not a getter which filters out items with [`isDeleted`](https://api.emberjs.com/ember-data/3.28/classes/Model/properties/isDeleted?anchor=isDeleted) set to true could fix that? It works in my head but maybe it's not so simple in reality :smile:.

Reply:
Ok I think I managed to do something about it, I'll push, don't hesitate to tell me if it's what you had in mind or not ! :sweat_smile: